src/Orleans/Core/GrainStateStorageBridge.cs

```diff
@@ -12,10 +12,11 @@ namespace Orleans.Core
     internal class GrainStateStorageBridge : IStorage
     {
         private readonly IStorageProvider store;
-        private readonly Grain grain;
+        private readonly IStatefulGrain grain;
```

Comment:
The ```Grain``` now does not possess state, so they are needed in order to have access to both ```grain.GrainState``` and  ```baseGrain.GrainReference``` properties.